src/lib/taggingUtils.ts

```diff
@@ -63,8 +82,32 @@ const taggingUtils = {
     return Object.keys(tagObject).map(tagKey => this.buildTag(tagKey, tagObject[tagKey]));
   },
 
-  generateCustomTags(annotationList: string[] = [], useFallback = false) {
-    return annotationList.map(el => this.buildTag(ANNOTATION_LABEL, el, useFallback));
+  /**
+   * Generate the list of tags out of a list of annotations
+   *
+   * Note: When generating the tags for a search, the tags will
+   *       include the fallback option, if it is required
+   * @param annotationList The list of annotations to build tags for
+   * @param isSearch Flag indicating if the tags are used for a search or not
+   * @returns A list of tags
+   */
+  generateCustomTags(annotationList: string[] = [], isSearch = false) {
+    return annotationList.map(annotation => {
+      const original = this.buildTag(ANNOTATION_LABEL, annotation, false);
+      if (!isSearch) {
+        return original;
+      }
+      /*
+      Original JS SDK implementation was inconsistent in lowercasing/uppercasing
+      some escaped characters, so we query for both versions by combining
+      both version into an OR query
+      */
+      const fallback = this.buildTag(ANNOTATION_LABEL, annotation, true);
+      if (original !== fallback) {
+        return `(${original},${fallback})`;
+      }
+      return original;
+    });
```

Comment:
To be honest - I am not a fan of flags - they make it super hard to understand stuff in a global perspective at least in my eyes. Mostly they are solvable with structural patterns - like in this case. Here you could simply split this method into 2. This also fits Single responsibility of SOLID better.

Also this method is super confusing for me due to the nested stuff. Also it could be dried out with some small changes and make use of the logic in `generateFhirVersionTag`, since they are not really different. (You could for example make use of [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set))